commit 4a715cc49d798e9b9f160c958c49239048ba4c69
Author: bertin <macportsuser@macports.org>
Date:   Sun May 25 22:22:15 2025 +0200

    /opt/local/site-ports/sysutils/MacPorts/files/patch-parallel-xz.diff

diff --git a/src/package1.0/portarchive.tcl b/src/package1.0/portarchive.tcl
index 92e8ce755..d5007b433 100644
--- a/src/package1.0/portarchive.tcl
+++ b/src/package1.0/portarchive.tcl
@@ -46,7 +46,8 @@ set_ui_prefix
 
 proc portarchive::archive_command_setup {location archive.type} {
     global archive.env archive.cmd archive.pre_args archive.args \
-           archive.post_args portarchive_hfscompression
+           archive.post_args portarchive_hfscompression \
+           build.jobs prefix
     set archive.env {}
     set archive.cmd {}
     set archive.pre_args {}
@@ -97,6 +98,7 @@ proc portarchive::archive_command_setup {location archive.type} {
                 set archive.cmd "$tar"
                 set archive.pre_args {-cvf}
                 if {[regexp {z2?$} ${archive.type}]} {
+                    set extraargs ""
                     if {[regexp {bz2?$} ${archive.type}]} {
                         if {![catch {binaryInPath lbzip2}]} {
                             set gzip "lbzip2"
@@ -111,6 +113,12 @@ proc portarchive::archive_command_setup {location archive.type} {
                         set level ""
                     } elseif {[regexp {xz$} ${archive.type}]} {
                         set gzip "xz"
+                        if {![catch {binaryInPath xz}]} {
+                            set gzip "xz"
+                        } else {
+                            set gzip "$prefix/sbin/xz"
+                        }
+                        set extraargs "--threads=${build.jobs}"
                         set level 6
                     } else {
                         set gzip "gzip"
@@ -122,9 +130,9 @@ proc portarchive::archive_command_setup {location archive.type} {
                         set hint ""
                     }
                     if {[catch {set gzip [findBinary $gzip $hint]} errmsg] == 0} {
-                        ui_debug "Using $gzip"
+                        ui_debug "Using $gzip -c$level $extraargs"
                         set archive.args {- .}
-                        set archive.post_args "| $gzip -c$level > [shellescape ${location}]"
+                        set archive.post_args "| $gzip -c$level $extraargs > [shellescape ${location}]"
                     } else {
                         ui_debug $errmsg
                         return -code error "No '$gzip' was found on this system!"
